src/csv.js

```diff
@@ -0,0 +1,34 @@
+import csv from 'minimal-csv-formatter'
+import _ from 'lodash/fp'
+
+let transformLabels = _.map(_.get('label'))
+
+export default ({
+  stream, // writable stream target stream
+  iterableData, // iterator for each page of an array of objects
+  // order list of which indicates the header label,
+  // display function for the field,
+  // and key of the record.
+  // [{ key: string, label: string, dispaly: funciton}...]
+  transform,
+  onWrite = _.noop, // function to intercept writing a page of records
+}) => {
+  stream.write(csv(transformLabels(transform)))
+  let cancel = false
+  let recordsWritten = 0
+
+  return {
+    promise: (async () => {
+      for await (let r of iterableData) {
+        if (cancel) break
+        stream.write(csv(_.map(t => t.display(r[t.key]), transform)))
```

Comment:
`r[t.key]` should probally use `_.get` so we can support nested fields (e.g. `LineItem.UnitPrice`). While you're at it, it would be trivial to make `display` optional by just defaulting it to `x => x` so it's more convenient to use. Display should also take the full row as the second parameter to be consistent with the rest of the contexture ecosystem (e.g. if you need to format conditionally based on the values of other fields)